packages/tools/viewer-alpha/src/viewer.ts

```diff
@@ -110,18 +160,121 @@ export class Viewer implements IDisposable {
         this._camera = new ArcRotateCamera("camera1", 0, 0, 1, Vector3.Zero(), this._details.scene);
         this._camera.attachControl();
         this._updateCamera(); // set default camera values
+        this._autoRotationBehavior = this._camera.getBehaviorByName("AutoRotation") as AutoRotationBehavior;
 
         // Load a default light, but ignore errors as the user might be immediately loading their own environment.
         this.loadEnvironmentAsync(undefined).catch(() => {});
 
         // TODO: render at least back ground. Maybe we can only run renderloop when a mesh is loaded. What to render until then?
-        this._engine.runRenderLoop(() => {
+        const render = () => {
             this._details.scene.render();
-        });
+            if (this.isAnimationPlaying) {
+                this.onAnimationProgressChanged.notifyObservers();
+                this._autoRotationBehavior.resetLastInteractionTime();
+            }
+        };
+
+        this._engine.runRenderLoop(render);
+        this._renderLoop = {
+            dispose: () => this._engine.stopRenderLoop(render),
+        };
 
         options?.onInitialized?.(this._details);
     }
 
+    /**
+     * The list of animation names for the currently loaded model.
+     */
+    public get animations(): readonly string[] {
+        return this._details.model?.animationGroups.map((group) => group.name) ?? [];
+    }
+
+    /**
+     * The currently selected animation index.
+     */
+    public get selectedAnimation(): number {
+        return this._selectedAnimation;
+    }
+
+    public set selectedAnimation(value: number) {
+        value = Math.round(Scalar.Clamp(value, -1, this.animations.length - 1));
+        if (value !== this._selectedAnimation) {
+            const startAnimation = this.isAnimationPlaying;
+            if (this._activeAnimation) {
+                this._activeAnimation?.stop();

```

Comment:
why the question mark if it is defined ?

Reply:
🤦🏼‍♂️